### PR TITLE
rtl8812au: init at 4.2.2

### PR DESCRIPTION
**Pull request by yochai**
Adds the 8812au driver.

### CODE REVIEW

File: pkgs/os-specific/linux/rtl8812au/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, kernel }:
+
+stdenv.mkDerivation {
+  name = "rtl8812au-${kernel.version}";
+  version = "4.2.2-1";
```
Comment (garbas) [inline pkgs/os-specific/linux/rtl8812au/default.nix]: i dont think this is used anywhere right?

Comment (garbas) [inline pkgs/os-specific/linux/rtl8812au/default.nix]: name should include version ... example: 
```
stdenv.mkDerivation rec {
  name = "rtl8812au-${kernel.version}-${version}";
  version = "4.2.2-1";
...
```

File: pkgs/os-specific/linux/rtl8812au/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    substituteInPlace ./Makefile --replace '$(MODDESTDIR)' "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
+  '';
+  
+  preInstall = ''
+    mkdir -p "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
+  '';
+   
+  meta = {
+    description = "Driver for Realtek 802.11ac, rtl8812au, provides the 8812au mod.";
+    homepage = "https://github.com/csssuf/rtl8812au";
```
Comment (garbas) [inline pkgs/os-specific/linux/rtl8812au/default.nix]: platform, maintainers and licence meta options would be nice to have here as well

**Comment by yochai** [comment]
Added advised changes.

**Comment by yochai** [comment]
It builds and works locally, as far as I can tell it fails due to grsecurity, I am not sure how to proceed.

**Pull request closed without merging**